sycl/source/detail/queue_impl.hpp

```diff
@@ -849,7 +859,14 @@ class queue_impl {
                     const std::shared_ptr<queue_impl> &SecondaryQueue,
                     bool CallerNeedsEvent, const detail::code_location &Loc,
                     bool IsTopCodeLoc, const SubmissionInfo &SubmitInfo);
-#endif
+#else  // __INTEL_PREVIEW_BREAKING_CHANGES
+  // template is needed for proper unit testing
+  template <typename HandlerType = handler>
```

Comment:
nit: I think we can remove the `finalizeHandlerPostProcess` function entirely and just move it's entire logic (expect for calling finalizeHandlerInOrder/finalizeHandlerOutOfOrder to a separate postProcess function. submit_impl would then look something like: 

```
auto event = finalizeHandler(Handler);
#ifndef __INTEL_PREVIEW_BREAKING_CHANGES
postProcess(SubmitInfo.PostProcessorFunc());
#endif
```

I've done a similar thing in https://github.com/intel/llvm/pull/18277 as going through finalizeHandlerPostProcess incurs some overhead even if PostProcessorFunc is null

Reply:
I have inlined the post-processing and removed the additional `fializeHandler` variant. I am reluctant to move the `KernelUsesAssert` to after the actual call to `finalizeHandler`, as it is the final call-site using the `Handler` object, so there's little telling if the handler is changed. I definitely want to avoid introducing a regression due to a reordering of operations that will be removed anyway.